src/nvim/os/os_defs.h

```diff
@@ -1,10 +1,137 @@
 #ifndef NVIM_OS_OS_DEFS_H
 #define NVIM_OS_OS_DEFS_H
 
+#include <ctype.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+// Note: Some systems need both string.h and strings.h (Savage). However,
+// some systems can't handle both, only use string.h in that case.
+#include <string.h>
+#if defined(HAVE_STRINGS_H) && !defined(NO_STRINGS_WITH_STRING_H)
+#  include <strings.h>
+#endif
+
+
 #ifdef WIN32
-# include "nvim/os/win_defs.h"
+#  include "nvim/os/win_defs.h"
+#else
+#  include "nvim/os/unix_defs.h"
+#endif
+
+#if !defined(HAVE_SYS_TIME_H) || defined(TIME_WITH_SYS_TIME)
+#  include <time.h>          // on some systems time.h should not be
+                             // included together with sys/time.h
+#endif
+
+#ifdef HAVE_SYS_TIME_H
+#  include <sys/time.h>
+#endif
+
+#include <signal.h>       // Valid on windows, but not recommended
+
+// Note: if MAXNAMLEN has the wrong value, you will get error messages
+// for not being able to open the swap file.
+#ifndef MAXNAMLEN
+#  ifdef NAME_MAX
+#    define MAXNAMLEN NAME_MAX
+#  else
+#    define MAXNAMLEN 255        // Seems to be the most supported value.
```

Comment:
Sorry, I should have been more clear, but I believe that C-style comments (`/* ... */`) must be used for macro definitions because
```c
char buf[MAXNAMELEN];
```
will be expanded like so:
```c
char buf[255      // ....];
```